### PR TITLE
feat(store-cache): add typescript datastore with support for composite keys

### PR DESCRIPTION
**Pull request by alvrs**
This PR adds a new client data store with support for composite keys, typed based on a given `StoreConfig`. The intention is for this to be used in combination with with the network package to mirror the state of a `Store` contract to the client. (Previously we used `recs` for this purpose, but `recs` does not support composite keys).

Todos in this PR:
- [x] add default values for each abi type
- [x] add mapping to primitive type for every abi type

Followup:
- [ ] add `recs` wrapper around a store cache (to be able to interact with contract ecs data via `recs` as before)
- [ ] update `setupMUDV2Network` to pipe contract data into store core rather than `recs`

### CODE REVIEW

File: packages/schema-type/src/typescript/utils/getAbiTypeDefaultValue.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { AbiTypeToPrimitiveType } from "../mappings";
+import { AbiTypeToDefaultValue } from "../mappings/AbiTypeToDefaultValue";
+
+/**
+ * Return a value corresponding to the Solidity default value for a given abi type
+ */
+export function getAbiTypeDefaultValue<T extends string>(abiType: T) {
+  // Remove fixed array lengths (eg map uint256[10] to uint256[])
+  const normalizedType = abiType.replace(/(\w+)\[\d+\]/g, "$1[]");
```
Comment (holic) [inline packages/schema-type/src/typescript/utils/getAbiTypeDefaultValue.ts:9]: let's hope no one names their array `$items` 🤓

File: packages/store-cache/package.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  },
+  "license": "MIT",
+  "type": "module",
+  "exports": {
+    ".": "./dist/index.js"
+  },
+  "types": "src/index.ts",
+  "scripts": {
+    "build": "tsup",
+    "clean": "rimraf dist",
```
Comment (holic) [inline packages/store-cache/package.json:19]: not blocking and not at all critical but I've been trying to keep the build scripts somewhat aligned so if we ever want to revisit turbo build steps, etc., we don't have to go through and audit all the build scripts again

**Review by holic — APPROVED** [review]
🎉🎉🎉🎉

**Review by holic — APPROVED** [review]
🎉🎉🎉🎉

**Review by dk1a — DISMISSED** [review]

**Pull request merged**